### ISSUE TITLE
DLL load failed with Python 3.9.5 - pywin32 v301 (virtual env)

### ISSUE THREAD

**Issue by teucer** [mhammond/pywin32#1709]
Note that issues in this repository are only for bugs or feature requests in the pywin32.

**If you need support or help using this package, please follow [these instructions](https://github.com/mhammond/pywin32/blob/master/README.md#support)** - support or help requests will be closed without comment.

For all bugs, please provide the following information.

* Expected behavior and actual behavior.

`pip install pywin32` should work

* Steps to reproduce the problem.

Execution of  `python -c "import win32api"` yields the following error:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: DLL load failed while importing win32api: The specified module could not be found.
```

* Version of Python and pywin32

  - Python: 3.9.5
  - pywin32: 301

* Details 

I have recently upgraded from Python 3.7.10 (pywin32 v300) to Python 3.9.5 (pywin32 v301).  I believe `win32\lib\pywin32_bootstrap.py` does not add properly DLLs to the path. When I manually edit the  PATH environment variable, the issue is resolved.

**Comment by samvanbrussel** [comment]
I also have this issue with Python 3.8.10. Rolling back to pywin32 v300 fixed my issues.

Reproduce in shell:

`python
import win32api
ImportError: DLL load failed while importing win32api: The specified procedure could not be found.`

After rolling back with `pip install pywin32==300`, there is no error any more.

**Comment by Ninjef** [comment]
TLDR; When a process tied to pywin32 **not equals** 301 tries to acquire its dlls, and the dlls for pywin32==301 are found first, you seem to get the DLL error instead of the process moving further down the path to find the correct dlls.

This issue caused me a great deal of stress over the past 4 days. I've narrowed down some details that I think would really help with fixing it.

I was running a process from within a python 3.8.5 conda environment (it's important to note that with conda environments, the PATH variable is still used for finding dlls even though python 3.8 did away with that). My PATH variable looked something like this:
```
C:\\Windows\\system32
C:\\Windows
C:\\Windows\\System32\\Wbem
C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\
C:\\Program Files\\Amazon\\cfn-bootstrap\\
C:\\ProgramData\\chocolatey\\bin
C:\\Program Files\\PowerShell\\7\\
C:\\Program Files\\OpenSSH-Win64
C:\\Users\\caveman\\Miniconda3
C:\\Users\\caveman\\Miniconda3\\Library\\mingw-w64\\bin
C:\\Users\\caveman\\Miniconda3\\Library\\usr\\bin
C:\\Users\\caveman\\Miniconda3\\Library\\bin
C:\\Users\\caveman\\Miniconda3\\Scripts
C:\\Users\\caveman\\AppData\\Local\\Microsoft\\WindowsApps
C:\\Users\\caveman\\AppData\\Local\\Programs\\Microsoft VS Code\\bin
D:\\Alteryx\\bin\\
D:\\Alteryx\\bin\\plugins
C:\\Users\\caveman\\AppData\\Roaming\\Stuff\\Thing_venv\\Lib\\site-packages\\pypiwin32_system32
C:\\Users\\caveman\\AppData\\Roaming\\Thing_venv\\lib\\site-packages\\pywin32_system32
```
Some stuff I was doing happened to run some code in jupyter_core where `import win32api` exists. That line would produce 
```
ImportError: DLL load failed while importing win32api: The specified procedure could not be found.
```
If the python process running jupyter_core was using anything other than pywin32==301. But I wanted that process to run pywin32==227. So I kept hitting the error.

Turns out that I had the dlls for 301 in `C:\\Windows\\system32` (from a manual attempt to alleviate a previous issue), and as you can see from my path, that is the first place that's checked for the dlls. Conversely, I had the dlls for 227 (which the process was looking for) in `C:\\Users\\jarnold\\AppData\\Roaming\\Stuff\\Thing_venv\\Lib\\site-packages\\pypiwin32_system32`. It seems that the process simply would say that the DLLs don't exist as soon as it found the **wrong** ones in `C:\\Windows\\system32`, instead of continuing the search down the PATH.

When I deleted the dlls for 301 in `C:\\Windows\\system32`, everything worked.

If I added the dlls for 300, 228, 227, or 225 into `C:\\Windows\\system32`, the error would not occur.

I'm not totally certain how the dll search happens, but it seems like something about the 301 dlls causes the search to immediately stop and throw the `DLL load failed` error.

I've confirmed this behavior on other, cleaner systems as well.

It'd be great to get a fix so that 301 or 302(?) can simply exist in the PATH and not kill the process if the dlls being hunted are for a different version.

Thanks! Hope this helps.

**Comment by yinkanchn** [comment]
python==3.9.6 pywin32==301

**Comment by mhammond** [comment]
> have this issue, rolling back with`python==3.8.5` `pywin32==300` work well

This will almost certainly mean a duplicate `pywintypes38.dll` or `pythoncom38.dll`

**Comment by mhammond** [comment]
I've added some relevant notes to https://github.com/mhammond/pywin32/blob/main/README.md#the-specified-procedure-could-not-be-found--entry-point-not-found-errors and other recent changes should help to resolve this kind of issue.
